app/components/Form/ProjectMilestoneReportForm.tsx

```diff
@@ -231,8 +234,17 @@ const ProjectMilestoneReportForm: React.FC<Props> = (props) => {
         >
           <FontAwesomeIcon icon={faPlusCircle} /> Add another milestone report
         </Button>
-
         {sortedMilestoneReports.map((node, index) => {
+          const localNewFormData = JSON.parse(JSON.stringify(node.newFormData));
```

Comment:
We've used this pattern when importing schema objects from their global definition, to make sure we weren't modifying the singleton in-place. In this case, the `node.newFormData` object is local to this file / form and is safe to modify in-place, and there is no concept of 'local' vs 'global'

```suggestion
          const formData = node.newFormData;
```

Reply:
We get `Cannot add property calculatedGrossAmountThisMilestone, object is not extensible` I think we need a deep copy here.

Reply:
I see. then object destructuring will do what you want:
```suggestion
                const formData = {...node.newFormData};
```